software/multiqc/main.nf

```diff
@@ -0,0 +1,32 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process MULTIQC {
+    tag "multiqc"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename: filename, options: params.options, publish_dir: getSoftwareName(task.process), publish_id: '') }
+
+    conda(params.enable_conda ? "bioconda::multiqc=1.9" : null)
+    container "quay.io/biocontainers/multiqc:1.9--pyh9f0ad1d_0"
+
+    input:
+    path multiqc_config
+    path multiqc_custom_config
+    path('fastqc/*')
+
+    output:
+    path "*multiqc_report.html", emit: report
+    path "*_data", emit: data
+    path "*_plots", optional: true, emit: plots
+
+    script:
+    def custom_config = params.multiqc_config ? "--config $multiqc_custom_config" : ''
```

Comment:
```suggestion
```

Reply:
Not sure we need this for a generic MultiQC modules @ewels ? Also, means whoever is using this module will need to use both `params.multiqc_config` and `$multiqc_custom_config`. We could leave it in but it is quite nf-core pipeline template specific and the main issue here is the ability of Nextflow to handle optional inputs.

Reply:
Cool, for the time being I've accepted this suggestion.